### PR TITLE
Rules order in policy showing drift during apply

### PR DESCRIPTION
**Pull request by jacklongsd**

### CODE REVIEW

File: sysdig/resource_sysdig_secure_custom_policy_test.go
Comment (ombellare) [inline sysdig/resource_sysdig_secure_custom_policy_test.go:36]: Nit: can we keep this as `rText()` for consistency as all the other tests already use it? Or you would need to change all tests to use your new variable

File: sysdig/resource_sysdig_secure_custom_policy.go
Comment (ombellare) [inline sysdig/resource_sysdig_secure_custom_policy.go:141]: Can we remove the Printfs added for debugging in this file please?

File: sysdig/resource_sysdig_secure_custom_policy.go
Comment (ombellare) [inline sysdig/resource_sysdig_secure_custom_policy.go:144]: Nit: `if !arePolicyRulesEquivalent(currentRules, newRules)` to remove the extra variable

File: sysdig/resource_sysdig_secure_custom_policy.go
Comment (ombellare) [inline sysdig/resource_sysdig_secure_custom_policy.go:143]: Can we use `reflect.DeepEqual` provided by Go for checking equality of maps? (https://techwasti.com/the-best-way-to-check-the-equality-of-two-maps-is-in-go). It would be tested for all edge cases and reduce the code that we need to maintain

**Review by ombellare — COMMENTED** [review]
Suggested some improvements. Overall LGTM

File: sysdig/resource_sysdig_secure_custom_policy.go
Comment (jacklongsd) [reply sysdig/resource_sysdig_secure_custom_policy.go:143]: I tried using that at first, took me forever to realize it wasn't working. The problem is the data structure is a slice of maps, and when doing reflect.DeepEquals it cares about the order of elements which is exactly what I am trying to avoid.

File: sysdig/resource_sysdig_secure_custom_policy_test.go
Comment (jacklongsd) [reply sysdig/resource_sysdig_secure_custom_policy_test.go:36]: rText() generates a random string to make each policy different. The purpose of the test is to create a policy, modify it's rules order and not see drift on tf plan. Need the names to be the same to achieve this

**Review by ombellare — APPROVED** [review]
LGTM! Makes sense to get one more team approval before merging 👍

**Pull request merged**
